runtime/doc/repeat.txt

```diff
@@ -172,9 +172,13 @@ Using Vim scripts					*using-scripts*
 For writing a Vim script, see chapter 41 of the user manual |usr_41.txt|.
 
 					*:so* *:source* *load-vim-script*
-:so[urce] {file}	Runs |Ex| commands or Lua code (".lua" files) read
-			from {file}.
+:so[urce] [file]	Runs |Ex| commands or Lua code (".lua" files) read
+			from [file].
 			Triggers the |SourcePre| autocommand.
+
+			When [file] is omitted, the contents of the current
+			buffer are executed instead.
```

Comment:
I've rebased this on top of the latest changes on master, though it also seems that `s:` variables don't work with anonymous `:source`s.

I was going to mention that here too, but it seems that they don't even work for `nvim_exec()` (which uses a similiar mechanism of using `SID_STR`), even though the docs says otherwise:

```
Unlike |nvim_command()| this function supports heredocs,
script-scope (s:), etc.
```

It seems like it's meant to be fixed in a follow up of #11500, so probably best to not mention it?

Reply:
hmm you mean they don't work at all, or more like anon `:%source` accesses `s:` in a different scope from when sourcing the file directly? (which is what I would expect)

Reply:
@bfredl 

```vim
let s:var = 10
echo s:var
```

works with `:so %` but not `:so`

Reply:
(also in retrospect, it might also be worth extending `nvim_exec()` to work with line continuations too? `get_str_line()` doesn't support them either)

Reply:
> also in retrospect, it might also be worth extending nvim_exec() to work with line continuations too? get_str_line() doesn't support them either

Yes, we should do that as well.